### PR TITLE
recipes: absent: Bump version 0.3.0 and cleanups

### PR DESCRIPTION
**Pull request by rvarago**
Specify library name and version:  **absent/0.3.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/absent/all/conanfile.py [hunk 1]
```diff
@@ -19,36 +19,29 @@ class AbsentConan(ConanFile):
...
+        elif not self._supports_cpp17():
             raise ConanInvalidConfiguration("Absent requires C++17 support")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
-    def build(self):
-        cmake = self._configure_cmake()
```
Comment (SSE4) [inline recipes/absent/all/conanfile.py:49]: why remove build?

Comment (rvarago) [reply recipes/absent/all/conanfile.py:49]: Because it's a header-only library. So, I thought that the build step was unnecessary. Should I bring it back?

Comment (SSE4) [reply recipes/absent/all/conanfile.py:49]: no, I think this is not needed in this case. I wonder why did we have it all.

File: recipes/absent/all/conanfile.py [hunk 2]
```diff
@@ -19,36 +19,29 @@ class AbsentConan(ConanFile):
     def _source_subfolder(self):
         return "source_subfolder"
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
```
Comment (uilianries) [inline recipes/absent/all/conanfile.py:24]: Why did you remove build_subfolder? The propose here is keeping all build stuff separated from the source files.

File: recipes/absent/all/conanfile.py [hunk 3]
```diff
@@ -19,36 +19,29 @@ class AbsentConan(ConanFile):
...
             raise ConanInvalidConfiguration("Absent requires C++17 support")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
-    def build(self):
-        cmake = self._configure_cmake()
-        cmake.build()
```
Comment (uilianries) [inline recipes/absent/all/conanfile.py:50]: OKay, now I understand why self.build_subfolder is no longer required :)

File: recipes/absent/all/conanfile.py (hunk 2 above)
Comment (rvarago) [reply recipes/absent/all/conanfile.py:24]: Because the library is header-only so I thought the build step and things related wouldn't be needed.

However, there's a WARN in the logs about not having the build step defined.

What should be the recommended way? Should I bring the build step back?

Comment (jgsogo) [reply recipes/absent/all/conanfile.py:24]: If it doesn't require a build step, it doesn't require it 👍  maybe we should show that warning in Conan.

Comment (rvarago) [reply recipes/absent/all/conanfile.py:24]: Got it.
Thanks for clarifying that.
